### PR TITLE
[IMP] fieldservice_substatus: Track Visibility + Notifications

### PR DESCRIPTION
**Pull request by osi-scampbell**
This PR tracks the visibility of sub-status changes and send a notification to followers similar to how Helpdesk Tickets send notifications to their followers for Stage Changes

### CODE REVIEW

File: fieldservice_substatus/data/mail_data.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo><data noupdate="1">
```
Comment (max3903) [inline fieldservice_substatus/data/mail_data.xml]:
```suggestion
<odoo noupdate="1">
```

File: fieldservice_substatus/data/mail_data.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <record id="fso_substatus_changed" model="mail.message.subtype">
+        <field name="name">Substatus Changed</field>
+        <field name="sequence">10</field>
+        <field name="res_model">fsm.order</field>
+        <field name="default" eval="False"/>
+        <field name="internal" eval="True"/>
+        <field name="description">Substatus Changed</field>
+    </record>
+    
+</data>
```
Comment (max3903) [inline fieldservice_substatus/data/mail_data.xml]:
```suggestion
```

**Review by max3903 — APPROVED** [review]
Minor changes

**Comment by osi-scampbell** [comment]
@bodedra @murtuzasaleh Hey guys, can I get a review on this PR please? Thanks!

**Review by murtuzasaleh — APPROVED** [review]
Code Review LGTM.

**Review by dreispt — APPROVED** [review]

**Pull request merged**